wallet/reference/non-evm-apis/starknet-snap-api.md

```diff
@@ -0,0 +1,867 @@
+- `contractFuncName`: `string` - Target function name of the contract.
+- `contractCallData`: `string` - (Optional) Call data for the target function with `,` as a separator.
+- `senderAddress`: `string` - Address of the sender.
+- `maxFee`: `string` - (Optional) Maximum gas fee allowed from the sender.
+  If not specified, the maximum fee is automatically calculated.
+- `chainId`: `string` - (Optional) ID of the target Starknet network.
+  The default is the Starknet Sepolia testnet.
+
+#### Returns
+
+The response from Starknet's `gateway/add_transaction` API endpoint.
+
+#### Example
+
+<Tabs>
+<TabItem value="Request">
+
+```js
+await getEip6963Provider.request({
+  method: "wallet_invokeSnap",
+  params: {
+    snapId: "npm:@consensys/starknet-snap",
+    request: {
+      method: "starkNet_sendTransaction",
+      params: {
+        contractAddress: "0x5B38Da6a701c568545dCfcB03FcB875f56beddC4",
+        contractFuncName: "transfer",
+        contractCallData: "0x456...,0x789...,100",
+        senderAddress: "0xb60e8dd61c5d32be8058bb8eb970870f07233155",
+        maxFee: "1000000000000000",
+        "chainId": "0x534e5f5345504f4c4941"
+      },
+    },
+  },
+})
+```
+
+</TabItem>
+<TabItem value="Result">
+
+```json
+{
+  "transaction_hash": "0x05a56e2d52c817161883f50c441c3228cfe54d9f84b5b5b8b1c8b8e0e6f7e6d8"
+}
+```
+
+</TabItem>
+</Tabs>
+
+### `starkNet_signMessage`
+
+:::note
+This method is supported by the `get-starknet` library.
+:::
+
+Signs a typed data message.
+
+#### Parameters
+
+- `typedDataMessage`: `string` - JSON representation of the typed data to be signed.
+- `signerAddress`: `string` - Address of the signer.
+- `chainId`: `string` - (Optional) ID of the target Starknet network.
+  The default is the Starknet Sepolia testnet.
+
+#### Returns
+
+The signed hash of typed data.
+
+#### Example
+
+<Tabs>
+<TabItem value="Request">
+
+```js
+await getEip6963Provider.request({
+  method: "wallet_invokeSnap",
+  params: {
+    snapId: "npm:@consensys/starknet-snap",
+    request: {
+      method: "starkNet_signMessage",
+      params: {
+        typedDataMessage: "{ ... }",
+        signerAddress: "0xb60e8dd61c5d32be8058bb8eb970870f07233155",
+        "chainId": "0x534e5f5345504f4c4941"
+      },
+    },
+  },
+})
+```
+
+</TabItem>
+<TabItem value="Result">
+
+```json
+"1234567890,9876543210"
+```
+
+</TabItem>
+</Tabs>
+
+### `starkNet_upgradeAccContract`
+
+:::note
+
+This method is supported by the `get-starknet` library.
+
+:::
+
+Upgrades an account contract.
+
+#### Parameters
+
+- `contractAddress`: `string` - Address of the target contract.
+- `maxFee`: `string` - (Optional) Maximum gas fee allowed from the sender.
+  If not specified, the maximum fee is automatically calculated.
+- `chainId`: `string` - (Optional) ID of the target Starknet network.
+  The default is the Starknet Sepolia testnet.
+
+#### Returns
+
+The response from Starknet's `gateway/call_contract` API endpoint.
```

Comment:
Since the `feeder_gateway`/`gateway` has been [deprecated](https://community.starknet.io/t/feeder-gateway-deprecation/100233), should this be updated?